db/system_keyspace.cc

```diff
@@ -1438,14 +1438,21 @@ future<> system_keyspace::build_bootstrap_info() {
     });
 }
 
-future<> system_keyspace::setup(sharded<netw::messaging_service>& ms) {
+future<> system_keyspace::setup_before_commitlog() {
+    assert(this_shard_id() == 0);
+
+    // It's safe to load truncation records here, since
+    // all the writes to the table are followed by flushes.
+    co_await cache_truncation_record();
```

Comment:
Just pull this out of system_keyspace.cc and put in the needed place on main(), please

Reply:
> Just pull this out of system_keyspace.cc and put in the needed place on main(), please

Yes, done.